src/Uno.UI/KeyboardRectProvider.Android.cs

```diff
@@ -65,20 +66,25 @@ public void Stop()
 		/// </summary>
 		void ViewTreeObserver.IOnGlobalLayoutListener.OnGlobalLayout()
 		{
-			var screenSize = new Point();
-			_activity.WindowManager.DefaultDisplay.GetSize(screenSize);
+			var metrics = new DisplayMetrics();
+			_activity.WindowManager.DefaultDisplay.GetMetrics(metrics);
+
+			var realMetrics = new DisplayMetrics();
+			_activity.WindowManager.DefaultDisplay.GetRealMetrics(realMetrics);
 
 			var popupRect = new Rect();
 			_popupView.GetWindowVisibleDisplayFrame(popupRect);
 
+			// We are only interested in the height above the navigation bar. If the navigation bar visibility changes, it will also raise OnGlobalLayout.
+			// While the keyboard is visible the navigation bar will always also be visible (enabling the close keyboard button).
 			var keyboardRect = new Rect(
 				0,
-				popupRect.Bottom,
-				screenSize.X,
-				screenSize.Y
+				Math.Min(popupRect.Bottom, metrics.HeightPixels),
+				realMetrics.WidthPixels,
+				realMetrics.HeightPixels
 			);
 
 			_onKeyboardRectChanged?.Invoke(keyboardRect);
 		}
```

Comment:
It seems like you're no longer calling RaiseConfigurationChanged() when OnGlobalLayout() is called, do you not need that any more?

Reply:
No because OnSystemUiVisibilityChangeListener handles it